### PR TITLE
142 - dataset ingestion from slate-scratch

### LINKED ISSUE #142: Ingest from Slate-scratch
This is a follow up of ticket #99. The intention is to enable the dataset ingestion from the Slate-Scratch filesystem.

### PR DESCRIPTION
**Pull request by ri-pandey**
**Description**

Dataset ingestion from Slate-scratch.

**Related Issue(s)**

Closes #142

**Changes Made**

List the main changes made in this PR. Be as specific as possible.

- [x] Feature added
- [ ] Bug fixed
- [x] Code refactored
- [x] Documentation updated
- [ ] Other changes: [describe]

**Checklist**

Before submitting this PR, please make sure that:

- [x] Your code passes linting and coding style checks.
- [x] Documentation has been updated to reflect the changes.
- [x] You have reviewed your own code and resolved any merge conflicts.
- [x] You have requested a review from at least one team member.
- [x] Any relevant issue(s) have been linked to this PR.

### CODE REVIEW

File: api/src/routes/fs.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+  req.query.path = p;
+  console.log('req.query.path: ', req.query.path);
+  console.log('next()');
+  next();
+}
+
+const get_mount_dir = (req) => {
+  const base_dir_key = getBaseDirKey(req);
+  console.log('get_mount_dir(): base_dir:', base_dir_key);
```
Comment (deepakduggirala) [inline api/src/routes/fs.js:79]: either use logger.debug or disable the console.log

File: api/src/routes/fs.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    // const query_path = req.query.path;
+
+    // const base_dir = Object.values(config.filesystem.base_dir).filter((dir)
+    // => req.query.path.startsWith(dir))[0]; console.log('base_dir: ',
+    // base_dir);
+    //
+    const mounted_search_dir = get_mounted_search_dir(req);
+    console.log('mounted_search_dir: ', mounted_search_dir);
+
+    if (!fs.existsSync(mounted_search_dir)) {
```
Comment (deepakduggirala) [inline api/src/routes/fs.js:138]: please use async version of fs

File: api/src/routes/fs.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      res.json([]);
+      return;
+    }
+
+    const files = fs.readdirSync(mounted_search_dir, {
```
Comment (deepakduggirala) [inline api/src/routes/fs.js:143]: async

**Comment by ri-pandey** [comment]
@deepakduggirala Updated as per last feedback.

**Review by deepakduggirala — APPROVED** [review]

**Review by deepakduggirala — APPROVED** [review]

**Pull request merged**